src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
```

Comment:
Actually before you even do that, consider moving the check into `PathHelper.GetSecretsPath(IFileProvider)`. As I explained before, it will actually never blow up here, but `PathHelper.GetSecretsPath(IFileProvider)` is public surface and the way it is right now it can NRE in the first line.